ansible/roles/network/tasks/main.yml

```diff
@@ -39,16 +43,86 @@
     dest: /boot/network.ini-sample
   tags:
     - touches_boot_partition
+  when: ansible_distribution_major_version|int != 9
 
 - name: Activate network manager
   command: systemctl enable screenly-net-manager.service chdir=/etc/systemd/system
+  when: ansible_distribution_major_version|int != 9
 
 - name: Activate network watchdog
   command: systemctl enable screenly-net-watchdog.timer chdir=/etc/systemd/system
+  when: ansible_distribution_major_version|int != 9
 
 - name: Install cronjob for watchdog
   cron:
     name: Screenly Net Watchdog
     minute: 0
     job: /usr/bin/python /usr/sbin/screenly_net_watchdog.py
     state: absent
+  when: ansible_distribution_major_version|int != 9
+
+
+# Use resin-wifi-connect if Stretch
+
+- name: Install NetworkManager
+  apt:
+    name: network-manager
+    state: present
+  when: ansible_distribution_major_version|int == 9
+
+- name: Disable network manager
+  command: systemctl disable screenly-net-manager.service
+  when: ansible_distribution_major_version|int == 9
+
+- name: Disable network watchdog
+  command: systemctl disable screenly-net-watchdog.timer
+  when: ansible_distribution_major_version|int == 9
+
+- name: Stop dhcpcd
+  command: systemctl stop dhcpcd
+  when: ansible_distribution_major_version|int == 9
+
+- name: Disable dhcpcd
+  command: systemctl disable dhcpcd
+  when: ansible_distribution_major_version|int == 9
+
+- name: Activate NetworkManager
+  command: systemctl enable NetworkManager
+  when: ansible_distribution_major_version|int == 9
+
+- name: Download resin-wifi-connect release
+  get_url:
+    url: https://github.com/resin-io/resin-wifi-connect/releases/download/v3.0.4/wifi-connect-v3.0.4-linux-rpi.tar.gz
+    dest: /home/pi/resin-wifi-connect.tar.gz
+  when: ansible_distribution_major_version|int == 9
```

Comment:
You should add a secondary when condition here to prevent the installer from running every time. Perhaps you could touch the file `/usr/local/share/wifi-connect/ui/.$VERSION`